Timeline two flaky/broken E2E scenarios

### Tasks

#939 did not help.
see [here](https://build.libelektra.org/blue/organizations/jenkins/PermaplanT/detail/master/478/pipeline) and also happened already multiple times elsewhere.
Not reproducable on my local machine on windows 10 wsl after running the timeline tests over 10 times, they always passed.

I did not use the bug template for this issue, as I dont think its a frontend bug or any bug that would benefit from the detailed infos of the template.

```
=========================== short test summary info ============================

FAILED steps/test_timeline.py::test_unhide_a_plant_by_changing_the_map_date - AssertionError: Locator expected to be visible

Actual value: None 

Call log:

LocatorAssertions.to_be_visible with timeout 30000ms

waiting for get_by_role("heading", name="Tomato (Solanum lycopersicum)")
```

--- a/doc/changelog.md
+++ b/doc/changelog.md
@@ -59,7 +59,6 @@ Syntax: `- short text describing the change _(Your Name)_`
 - _()_
 - _()_
 - _()_
-- _()_
 - Fix regression where Full plant names are not shown _(Moritz)_
 - _()_
 - _()_
@@ -74,7 +73,7 @@ Syntax: `- short text describing the change _(Your Name)_`
 - _()_
 - _()_
 - _()_
-- _()_
+- E2E: Increase e2e timeline/baselayer test stability, replace timeouts with sleep #1010 _(4ydan)_
 
 ## 0.3.5 - 06.11.2023 (320 commits)
 

--- a/e2e/pages/abstract_page.py
+++ b/e2e/pages/abstract_page.py
@@ -1,4 +1,5 @@
 import re
+import time
 from abc import ABC
 from e2e.pages.constants import E2E_TIMEOUT
 from playwright.sync_api import Page, expect
@@ -26,7 +27,7 @@ class AbstractPage(ABC):
         expect.set_options(timeout=E2E_TIMEOUT)
         response = self._page.goto(self.URL)
         assert response.status == 200
-        self._page.wait_for_timeout(1000)
+        time.sleep(1)
         self.verify()
         self.click_english_translation()
 

--- a/e2e/pages/home.py
+++ b/e2e/pages/home.py
@@ -28,7 +28,9 @@ class HomePage(AbstractPage):
         """
         Clicks the login button which
         navigates to the `LoginPage`.
+        Waits for the backend to be ready before doing so.
         """
+        self._page.get_by_test_id("login-button__is-loading").wait_for(state="detached")
         self._login_button.click()
 
     def click_logout_button(self):

--- a/e2e/pages/maps/planting.py
+++ b/e2e/pages/maps/planting.py
@@ -1,4 +1,5 @@
 import cv2
+import time
 import numpy as np
 from datetime import datetime, timedelta
 from playwright.sync_api import Page, expect, TimeoutError as PlaywrightTimeoutError
@@ -15,11 +16,20 @@ class MapPlantingPage(AbstractPage):
     def __init__(self, page: Page):
         self._page = page
         self._screenshot = None
+
         """ Selectors """
         # Navbar
         self._map_management_button = page.get_by_role("button", name="Maps")
+
         # Left side bar
+        self._added_date_state_idle = page.get_by_test_id(
+            "planting-attribute-edit-form__add-date-idle"
+        )
+        self._removed_on_state_idle = page.get_by_test_id(
+            "planting-attribute-edit-form__removed-on-idle"
+        )
         self._delete_plant_button = page.get_by_role("button", name="Delete Planting")
+
         # Layer visibility
         self._hide_plant_layer = page.get_by_test_id(
             "layer-list-item__plants-layer-visibility-icon"
@@ -27,16 +37,21 @@ class MapPlantingPage(AbstractPage):
         self._hide_base_layer = page.get_by_test_id(
             "layer-list-item__base-layer-visibility-icon"
         )
+
         # Canvas
         self._canvas = page.get_by_test_id("base-stage__canvas")
         self._close_selected_plant = self._canvas.get_by_role("button")
         self._map_date = page.get_by_label("Change map date")
+        self._timeline_idle = page.get_by_test_id("timeline__date-form-idle")
+
         # Top left section
         self._undo_button = page.get_by_test_id("map__undo-button")
         self._redo_button = page.get_by_test_id("map__redo-button")
+
         # Selected plant section
         self._plant_added_date = page.get_by_label("Added on")
         self._plant_removed_date = page.get_by_label("Removed on")
+
         # Plant layer
         self._base_layer_radio = page.get_by_test_id(
             "layer-list-item__base-layer-radio"
@@ -46,15 +61,25 @@ class MapPlantingPage(AbstractPage):
         )
         self._plant_search_icon = page.get_by_test_id("plant-search__icon-button")
         self._plant_search_input = page.get_by_test_id("plant-search__search-input")
+
         # Base layer
-        self._background_input = page.get_by_test_id(
+        self._background_image_file_path = page.get_by_test_id(
             "base-layer-right-toolbar__background-input"
+        )
+        self._background_image_file_path_idle = page.get_by_test_id(
+            "base-layer-attribute-edit-form__background-image-file-path-idle"
         )
         self._background_button = page.get_by_role("button", name="Choose an image")
         self._rotation_input = page.get_by_test_id(
             "base-layer-right-toolbar__rotation-input"
         )
+        self._rotation_input_idle = page.get_by_test_id(
+            "base-layer-attribute-edit-form__rotation-idle"
+        )
         self._scale_input = page.get_by_test_id("base-layer-right-toolbar__scale-input")
+        self._scale_input_idle = page.get_by_test_id(
+            "base-layer-attribute-edit-form__scale-idle"
+        )
 
     """ACTIONS"""
 
@@ -70,16 +95,21 @@ class MapPlantingPage(AbstractPage):
         """Changes the date by a given amount of days."""
         day = datetime.today() + timedelta(days=delta_days)
         self._map_date.fill(day.strftime("%Y-%m-%d"))
+        self._timeline_idle.wait_for()
 
     def change_plant_added_date_by_days(self, delta_days: int):
-        """Changes the date by a given amount of days."""
+        """Changes the date by a given amount of days and checks the spinner if possible."""
         day = datetime.today() + timedelta(days=delta_days)
         self._plant_added_date.fill(day.strftime("%Y-%m-%d"))
+        if delta_days < 0:
+            self._added_date_state_idle.wait_for()
 
     def change_plant_removed_date_by_days(self, delta_days: int):
-        """Changes the date by a given amount of days."""
+        """Changes the date by a given amount of days  and checks the spinner if possible."""
         day = datetime.today() + timedelta(days=delta_days)
         self._plant_removed_date.fill(day.strftime("%Y-%m-%d"))
+        if delta_days > 0:
+            self._removed_on_state_idle.wait_for()
 
     def fill_plant_search(self, plantname):
         """Clicks the search icon and types plantname into the plant search."""
@@ -88,21 +118,16 @@ class MapPlantingPage(AbstractPage):
     def fill_rotation(self, rotation):
         """Fills the rotation input according to rotation."""
         self._rotation_input.fill(rotation)
-        # Wait for input to be processed and saved
-        self._page.wait_for_timeout(2000)
+        self._rotation_input_idle.wait_for()
 
     def fill_scaling(self, scaling):
         """Fills the scaling input according to scaling."""
         self._scale_input.fill(scaling)
-        # Wait for input to be processed and saved
-        self._page.wait_for_timeout(2000)
+        self._scale_input_idle.wait_for()
 
     def close_tour(self):
         """
-        Currently you get greeted with the tour.
-        In case you rerun the tests locally or if this gets removed
-        it is wrapped, so it does not fail the tests.
-        The timeout is also lowered to not slow down the tests by a lot.
+        Closes the tour window and catches the exception if the tour window does not exist.
         """
         try:
             tour_close = self._page.get_by_label("Close Tour")
@@ -125,16 +150,16 @@ class MapPlantingPage(AbstractPage):
         self._page.get_by_test_id("plant-list-item__" + plant_name).click()
 
     def click_on_canvas_middle(self):
-        self._page.wait_for_timeout(2000)
         """Clicks in the middle of the canvas with a 300ms delay."""
+        time.sleep(1)
         box = self._canvas.bounding_box()
         self._page.mouse.click(
             box["x"] + box["width"] / 2, box["y"] + box["height"] / 2
         )
-        self._page.wait_for_timeout(200)
 
     def drag_select_box_over_canvas(self):
         """Drags a select box over 75% of the canvas from top left to bottom right"""
+        time.sleep(1)
         box = self._canvas.bounding_box()
         x = box["x"]
         y = box["y"]
@@ -154,7 +179,6 @@ class MapPlantingPage(AbstractPage):
         self._page.mouse.click(
             box["x"] + box["width"] / 4, box["y"] + box["height"] / 4
         )
-        self._page.wait_for_timeout(300)
 
     def click_delete(self):
         """Deletes a planting by clicks on the delete button."""
@@ -176,17 +200,15 @@ class MapPlantingPage(AbstractPage):
         """
         self._background_button.click()
         self._page.get_by_text("Birdie.jpg").click()
-        # Delay so image can be rendered on canvas
-        self._page.wait_for_timeout(2000)
+        self._background_image_file_path_idle.wait_for()
 
     def click_hide_base_layer(self):
         self._hide_base_layer.click()
 
     def click_hide_plant_layer(self):
         self._hide_plant_layer.click()
-        self._page.wait_for_timeout(5000)
 
-    def screenshot_canvas(self, timeout=10000):
+    def screenshot_canvas(self, timeout=3):
         """
         Takes a grayscale screenshot of the canvas.
         Has a default 300ms delay before taking the screenshot,
@@ -194,19 +216,21 @@ class MapPlantingPage(AbstractPage):
 
         Parameters
         ----------
-        timeout : int, optional, default=10000
+        timeout : int, optional, default=3 in seconds
         Timeout in ms before taking the screenshot.
 
         Returns
         -------
         opencv2 image object of the screenshot
         """
-        self._page.wait_for_timeout(timeout)
+        time.sleep(timeout)
         buffer = self._canvas.screenshot()
         self._screenshot = cv2.imdecode(
             np.frombuffer(buffer, dtype=np.uint8), cv2.IMREAD_GRAYSCALE
         )
         return self._screenshot
+
+    """ASSERTIONS"""
 
     def expect_canvas_equals_last_screenshot(self, test_name, rtol=0, atol=5):
         """
@@ -248,8 +272,6 @@ class MapPlantingPage(AbstractPage):
             cv2.imwrite("test-results/screenshots" + test_name + "-actual.png", actual)
             raise err
 
-    """ASSERTIONS"""
-
     def expect_plant_on_canvas(self, plant_name):
         """
         Confirms that the plant is on the canvas,
@@ -266,6 +288,7 @@ class MapPlantingPage(AbstractPage):
         by clicking in the middle of the canvas
         and making sure the delete button is not visible.
         """
+
         self.click_on_canvas_middle()
         expect(self._delete_plant_button).not_to_be_visible()
 
@@ -278,7 +301,7 @@ class MapPlantingPage(AbstractPage):
         expect(self._page.get_by_test_id("plant-search__empty-results")).to_be_visible()
 
     def expect_background_image(self, name):
-        expect(self._background_input).to_have_value(name)
+        expect(self._background_image_file_path).to_have_value(name)
 
     def expect_rotation_to_have_value(self, val):
         """Expects that the rotation is properly set."""

--- a/frontend/src/features/auth/components/LoginButton.tsx
+++ b/frontend/src/features/auth/components/LoginButton.tsx
@@ -33,7 +33,7 @@ export const LoginButton = () => {
       onClick={() => void auth.signinRedirect()}
     >
       {t('auth:log_in')}
-      <ConditionalLoadingSpinner show={auth.isLoading} />
+      <ConditionalLoadingSpinner show={auth.isLoading} data-testid="login-button__is-loading" />
     </SimpleButton>
   );
 };

--- a/frontend/src/features/map_planning/components/timeline/Timeline.tsx
+++ b/frontend/src/features/map_planning/components/timeline/Timeline.tsx
@@ -65,7 +65,12 @@ export function Timeline({ onSelectDate, defaultDate }: TimelineProps) {
       {submitState === 'loading' && (
         <CircleDottedIcon className="mb-3 mt-auto h-5 w-5 animate-spin text-secondary-400" />
       )}
-      {submitState === 'idle' && <CheckIcon className="mb-3 mt-auto h-5 w-5 text-primary-400" />}
+      {submitState === 'idle' && (
+        <CheckIcon
+          className="mb-3 mt-auto h-5 w-5 text-primary-400"
+          data-testid="timeline__date-form-idle"
+        />
+      )}
       {submitState === 'error' && (
         <span className="mb-3 mt-auto text-sm text-red-400">{t('timeline:date_error')}</span>
       )}

--- a/frontend/src/features/map_planning/layers/base/components/BaseLayerAttributeEditForm.tsx
+++ b/frontend/src/features/map_planning/layers/base/components/BaseLayerAttributeEditForm.tsx
@@ -110,7 +110,10 @@ export function BaseLayerAttributeEditForm({ onChange, isReadOnlyMode }: BaseLay
                   <CircleDottedIcon className="h-5 w-5 animate-spin text-secondary-400" />
                 ))}
               {!checkingImage && imageExists && pathSubmitState === 'idle' && (
-                <CheckIcon className="h-5 w-5 text-primary-400" />
+                <CheckIcon
+                  className="h-5 w-5 text-primary-400"
+                  data-testid="base-layer-attribute-edit-form__background-image-file-path-idle"
+                />
               )}
               {!checkingImage && !imageExists && pathSubmitState === 'idle' && (
                 <CrossIcon className="h-5 w-5 text-red-500" />
@@ -134,7 +137,12 @@ export function BaseLayerAttributeEditForm({ onChange, isReadOnlyMode }: BaseLay
             {rotationSubmitState === 'loading' && (
               <CircleDottedIcon className="h-5 w-5 animate-spin text-secondary-400" />
             )}
-            {rotationSubmitState === 'idle' && <CheckIcon className="h-5 w-5 text-primary-400" />}
+            {rotationSubmitState === 'idle' && (
+              <CheckIcon
+                className="h-5 w-5 text-primary-400"
+                data-testid="base-layer-attribute-edit-form__rotation-idle"
+              />
+            )}
           </span>
         }
         type="number"
@@ -152,7 +160,12 @@ export function BaseLayerAttributeEditForm({ onChange, isReadOnlyMode }: BaseLay
                 {scaleSubmitState === 'loading' && (
                   <CircleDottedIcon className="h-5 w-5 animate-spin text-secondary-400" />
                 )}
-                {scaleSubmitState === 'idle' && <CheckIcon className="h-5 w-5 text-primary-400" />}
+                {scaleSubmitState === 'idle' && (
+                  <CheckIcon
+                    className="h-5 w-5 text-primary-400"
+                    data-testid="base-layer-attribute-edit-form__scale-idle"
+                  />
+                )}
               </span>
             }
             type="number"

--- a/frontend/src/features/map_planning/layers/plant/components/PlantingAttributeEditForm.tsx
+++ b/frontend/src/features/map_planning/layers/plant/components/PlantingAttributeEditForm.tsx
@@ -171,7 +171,10 @@ export function PlantingAttributeEditForm({
           <CircleDottedIcon className="mb-3 mt-auto h-5 w-5 animate-spin text-secondary-400" />
         )}
         {addDateSubmitState === 'idle' && (
-          <CheckIcon className="mb-3 mt-auto h-5 w-5 text-primary-400" />
+          <CheckIcon
+            className="mb-3 mt-auto h-5 w-5 text-primary-400"
+            data-testid="planting-attribute-edit-form__add-date-idle"
+          />
         )}
       </div>
       <p className="pb-4 text-[0.8rem] text-neutral-400">
@@ -194,7 +197,10 @@ export function PlantingAttributeEditForm({
           <CircleDottedIcon className="mb-3 mt-auto h-5 w-5 animate-spin text-secondary-400" />
         )}
         {removeDateSubmitState === 'idle' && (
-          <CheckIcon className="mb-3 mt-auto h-5 w-5 text-primary-400" />
+          <CheckIcon
+            className="mb-3 mt-auto h-5 w-5 text-primary-400"
+            data-testid="planting-attribute-edit-form__removed-on-idle"
+          />
         )}
       </div>
 

